File: src/test/java/io/lsdconsulting/lsd/distributed/generator/diagram/label/LabelGeneratorMapShould.java
package io.lsdconsulting.lsd.distributed.generator.diagram.label;

import io.lsdconsulting.lsd.distributed.access.model.InterceptedInteraction;
import org.junit.jupiter.api.Test;

import static io.lsdconsulting.lsd.distributed.access.model.Type.*;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

class LabelGeneratorMapShould {

    private final LabelGeneratorMap labelGeneratorMap = new LabelGeneratorMap();

    @Test
    void generateRequestLabel() {
        InterceptedInteraction interaction = InterceptedInteraction.builder()
                .type(REQUEST)
                .httpMethod("GET")
                .path("path")
                .build();

        String result = labelGeneratorMap.generate(interaction);

        assertThat(result, is("GET path"));
    }

    @Test
    void generateResponseLabel() {
        InterceptedInteraction interaction = InterceptedInteraction.builder()
                .type(RESPONSE)
                .httpStatus("OK")
                .elapsedTime(20L)
                .build();

        String result = labelGeneratorMap.generate(interaction);

        assertThat(result, is("sync OK response (20 ms)"));
    }

    @Test
    void generatePublishLabel() {
        InterceptedInteraction interaction = InterceptedInteraction.builder().type(PUBLISH).build();

        String result = labelGeneratorMap.generate(interaction);

        assertThat(result, is("publish event"));
    }

    @Test
    void generateConsumeLabel() {
        InterceptedInteraction interaction = InterceptedInteraction.builder().type(CONSUME).build();

        String result = labelGeneratorMap.generate(interaction);

        assertThat(result, is("consume message"));
    }
}

File: src/test/java/io/lsdconsulting/lsd/distributed/generator/diagram/label/LabelGeneratorMapShould.kt
package io.lsdconsulting.lsd.distributed.generator.diagram.label

import io.lsdconsulting.lsd.distributed.access.model.InteractionType
import io.lsdconsulting.lsd.distributed.access.model.InterceptedInteraction
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.junit.jupiter.api.Test
import java.time.ZoneId
import java.time.ZonedDateTime

internal class LabelGeneratorMapShould {

    private val underTest = LabelGeneratorMap()

    @Test
    fun `generate request label`() {
        val interaction = InterceptedInteraction(traceId = "", interactionType = InteractionType.REQUEST, httpMethod = "GET", path = "path", elapsedTime = 0, createdAt = ZonedDateTime.now(ZoneId.of("UTC")))
        val result = underTest.generate(interaction)
        assertThat(result, `is`("GET path"))
    }

    @Test
    fun `generate response label`() {
        val interaction = InterceptedInteraction(traceId = "", interactionType = InteractionType.RESPONSE, httpStatus = "OK", elapsedTime = 20L, createdAt = ZonedDateTime.now(ZoneId.of("UTC")))
        val result = underTest.generate(interaction)
        assertThat(result, `is`("sync OK response (20 ms)"))
    }

    @Test
    fun `generate publish label`() {
        val interaction = InterceptedInteraction(traceId = "", interactionType = InteractionType.PUBLISH, elapsedTime = 0, createdAt = ZonedDateTime.now(ZoneId.of("UTC")))
        val result = underTest.generate(interaction)
        assertThat(result, `is`("publish event"))
    }

    @Test
    fun `generate consume label`() {
        val interaction = InterceptedInteraction(traceId = "", interactionType = InteractionType.CONSUME, elapsedTime = 0, createdAt = ZonedDateTime.now(ZoneId.of("UTC")))
        val result = underTest.generate(interaction)
        assertThat(result, `is`("consume message"))
    }
}
